engines/causal-decoder/engine.js

```diff
@@ -0,0 +1,124 @@
+import { execFile } from "node:child_process";
+import util from "node:util";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+
+const execFileP = util.promisify(execFile);
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const desc = `
+Causal Decoder is a transformer decoder-only model fine-tuned on Qwen2.5-1.5B, which is an open-source small LLM with only 1.5 billion parameters. The model was fine-tuned using PyTorch. It is not a chat model, so it only works properly when the input is of the form 'more x leads to more y' etc. This model is in the process of development.
+`;
+
+
+class Engine {
+  static description() {
+    return desc;
+  }
+
+  static supportedModes() {
+    return ["cld"];
```

Comment:
In your supported modes, do a check to see if you can find/run python.  If you can then return CLD, if you can't return an empty array.  This will disable your engine if python isn't installed properly on the hostmachine